src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -20,6 +20,8 @@ internal static partial class CpuMathUtils
         // The count of bytes in a 32-bit float, corresponding to _cbAlign in AlignedArray
         private const int FloatAlignment = 4;
 
+        private const int MinVectorSize = 16;
```

Comment:
Maybe this should be called `MinVectorSizeX86`? Arm, for example, actually supports a `Vector64<>` type.

Reply:
Ah, I see. This is saying only use this code path for 16 elements arrays. The name is slightly confusing.

Reply:
changed to ```MinInputSize```